### PR TITLE
Use liner instead of ssh/terminal.

### PR DESCRIPTION
**Pull request by mberhault**
Simpler setup, still has history, ctrl-D, tab completion, etc..
Only change for now is that it ctrl-C abort the current line (it was
previously ignored entirely and the signal was not catchable).

### CODE REVIEW

**Comment by tamird** [comment]
Needs go generate

**Comment by MBerhault** [comment]
yup. Done.

File: cli/sql.go [hunk 1]
```diff
@@ -117,43 +105,26 @@ func runTerm(cmd *cobra.Command, args []string) {
...
-	}
+	liner := liner.NewLiner()
 	defer func() {
-		_ = terminal.Restore(0, oldState)
+		_ = liner.Close()
 	}()
 
-	term := terminal.NewTerminal(readWriter, "> ")
 	for {
-		line, err := term.ReadLine()
```
Comment (tamird) [inline cli/sql.go]: why rename from `line` to `l`?

**Comment by tamird** [comment]
Not sure the extra dependency is worth it but LGTM after squashing

File: cli/sql.go (hunk 1 above)
Comment (MBerhault) [inline cli/sql.go]: line was too close to liner.

**Comment by tschottdorf** [comment]
Cool. LGTM. BTW, running a session from stdin would be useful. (`cockroach sql < file.sql`).

**Comment by MBerhault** [comment]
that already works, it's just really ugly since we don't print the commands
and the output isn't great to start with (eg: no delimiter between column
names and values).

$ cat foo.sql
show databases
set database = bar
show tables
select * from Foo
$ ./cockroach  sql --addr=localhost:10000 --certs=/store/cockroach/certs/
 < foo.sql
Database
bar
foo
OK
Table
Foo
k v


On Thu, Jul 30, 2015 at 6:36 AM, Tobias Schottdorf <notifications@github.com

**Comment by tschottdorf** [comment]
Ah, thanks for the heads up. I remember trying this at some point and not
being able to.

**Pull request merged**